pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -52,8 +58,14 @@ type predicateMetadata struct {
 	podBestEffort bool
 	podRequest    *schedulercache.Resource
 	podPorts      []*v1.ContainerPort
-	//key is a pod full name with the anti-affinity rules.
-	matchingAntiAffinityTerms map[string][]matchingPodAntiAffinityTerm
+
+	// A map of antiffinity terms' topology pairs to the pods'
+	// that can potentially match the affinity rules of the pod
+	topologyPairToAntiAffinityPods map[topologyPair][]*v1.Pod
+
+	// Reverse map for topologyPairToAntiAffinityPods to reduce deletion time
+	antiAffinityPodToTopologyPairs map[*v1.Pod][]topologyPair
```

Comment:
prefer not to use a pointer as key of map; `namespace/name` or `uid` is better

Reply:
+1

`==` semantics of pointer is somewhat confusing - although it works sometimes.

Reply:
We used pod pointer as key of map because in `topologyPairToAntiAffinityPods` we needed a reference to the actual instance of the pod as a value because we needed to access the list of the affinity terms of the pod which I was only available if we have a reference to the pod, however we used the pointer as a key in the reverse map just for symmetry, so we'd be using the same type in both maps.
Do you think we should change it to pod name in `antiAffinityPodToTopologyPairs` and leave it as pointer to pod in `topologyPairToAntiAffinityPods`?